pkg/sidecar/identity/member_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2021 ScyllaDB
+
+package identity
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestMember_GetSeeds(t *testing.T) {
+	createPodAndSvc := func(name, ip string, creationTimestamp time.Time) (*corev1.Pod, *corev1.Service) {
+		pod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: "namespace",
+				Labels: map[string]string{
+					naming.ClusterNameLabel: "my-cluster",
+				},
+				CreationTimestamp: metav1.NewTime(creationTimestamp),
+			},
+		}
+		svc := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: "namespace",
+			},
+			Spec: corev1.ServiceSpec{
+				ClusterIP: ip,
+			},
+		}
+		return pod, svc
+	}
+
+	now := time.Now()
+	firstPod, firstService := createPodAndSvc("pod-0", "1.1.1.1", now)
+	secondPod, secondService := createPodAndSvc("pod-1", "2.2.2.2", now.Add(time.Second))
+
+	ts := []struct {
+		Name         string
+		MemberName   string
+		MemberIP     string
+		Objects      []runtime.Object
+		ExpectedSeed string
+	}{
+		{
+			Name:         "bootstraps with itself when cluster is empty",
+			MemberName:   firstPod.Name,
+			MemberIP:     firstService.Spec.ClusterIP,
+			Objects:      []runtime.Object{firstPod, firstService},
+			ExpectedSeed: firstService.Spec.ClusterIP,
+		},
+		{
+			Name:         "bootstrap with any UN node",
+			MemberName:   firstPod.Name,
+			Objects:      []runtime.Object{firstPod, firstService, markPodReady(secondPod), secondService},
+			ExpectedSeed: secondService.Spec.ClusterIP,
+		},
+		{
+			Name:         "bootstrap with first created Pod when all are down",
+			MemberName:   firstPod.Name,
+			Objects:      []runtime.Object{firstPod, firstService, secondPod, secondService},
+			ExpectedSeed: secondService.Spec.ClusterIP,
+		},
+	}
```

Comment:
test case: no pod listed => errror

Reply:
don't you need at least 3 pods to test the sorting since it sorts only the "otherPods"

Reply:
test case: second pod is not ready but third is ready => third pod